specification/apimanagement/control-plane/Microsoft.ApiManagement/preview/2017-03-01/apimreports.json

```diff
@@ -0,0 +1,652 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApiManagementClient",
+    "description": "Use these REST APIs to get the analytics reports associated with your Azure API Management deployment.",
+    "version": "2017-03-01"
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{apimBaseUrl}",
+    "useSchemePrefix": false,
+    "positionInOperation": "first",
+    "parameters": [
+      {
+        "name": "apimBaseUrl",
+        "description": "The management endpoint of the API Management service, for example https://myapimservice.management.azure-api.net.",
+        "required": true,
+        "type": "string",
+        "in": "path",
+        "x-ms-skip-url-encoding": true
+      }
+    ]
+  },
+  "schemes": [
+    "https"
+  ],
+  "securityDefinitions": {
+    "apim_key": {
+      "type": "apiKey",
+      "name": "Authorization",
+      "in": "header"
+    }
+  },
+  "security": [
+    {
+      "apim_key": []
+    }
+  ],
+  "paths": {
+    "/reports/byApiId": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Reports_ListByApi",
```

Comment:
>Reports_ListByApi [](start = 24, length = 17)

Is there an PI parameter?

Reply:
Look at example ApiManagementGetReportsByApi. It generates report by api

---
In reply to: [163390995](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163390995) [](ancestors = 163390995)

Reply:
You mean they are ordered by API?

---
In reply to: [163392591](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163392591) [](ancestors = 163392591,163390995)

Reply:
these are analytics reports, which tell, how many calls to api succeeded failed etc for the provided filter range.

Reply:
What I mean, how would I choose which API I want to get lists from?

Reply:
I am not sure I understand what is the difference between ListByApi and ListByUser for example

---
In reply to: [163423669](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163423669) [](ancestors = 163423669,163392591,163390995)